### ISSUE TITLE
Success Criterion 2.1.2 - No Keyboard Trap - Level A

### ISSUE THREAD

**Issue by JJdeGroot** [w3c/matf#30]
WCAG2ICT guidance: https://www.w3.org/TR/wcag2ict-22/#no-keyboard-trap

Share your thoughts for applying to mobile apps as a comment below.

**Comment by JJdeGroot** [comment]
Discussed in today’s meeting.

<details>
<summary>12 February 2025</summary>

[Source](https://www.w3.org/2025/02/12-matf-minutes.html#8909)

~~~text
2.1.2 No Keyboard Trap

JJ: moving to no keyboard trap SC discussion

<Tori> I'd prefer something like "All functionality of the content is operable through a keyboard interface //or the standard mobile input// without requiring specific timings for individual keystrokes..."

<Tori> It needs a lot more iterations to get it right, but we need something way tighter if we do this. It needs to focus on the operable part of an accessibilty interface and not be generic

JJ: flutter issues could be flagged under 2.1.2

JJ: keyboard focus using a keyboard interface, keyboard interface could be replaced with accessibility interface and could replace keyboard focus with accessibility focus

Jamie: WCAG2ICT definition different

ACTION: Check why WCAG2ICT does not have Non-Interference section

JJ: WCAG2ICT removed conformance requirement 5 and replaced the word page with non-web document or software

<Tori> +1 to add back in Section 5 of WCAG

JJ: note 1, note 2 exit methods escape key, note 3 focus/keyboard traps

JJ: how well known to standard exit methods need to be known, e.g., android back gesture, iOS Z gesture

JJ: should this group address these exit gestures

JJ: long press not keyboard shortcut

<Joe_Humbert> switch access has a go back command

JJ: does there need to be visible exit button or do standard exit gestures work

ACTION: Add examples of accessibility interface exit functionality (Escape, "Go Back", Z-gesture, etc.)

pauljadam: want to build good bad examples of these

<JJ> [w3c/matf#30](https://github.com/w3c/matf/issues/30)

JJ: see if there are any comments on github

JJ: or move on to other SCs until we made a decision on keyboard and keyboard trap

Jamie: should be list keyboard interface definition issue

Jamie: need to different between SC that use just keyboard or keyboard interface

ACTION: Add SC's affected by "keyboard interface" to the issue for the definition - Jamie
~~~
</details>

- ACTION: Check why WCAG2ICT does not have Non-Interference section
- ACTION: Add examples of accessibility interface exit functionality (Escape, "Go Back", Z-gesture, etc.)
- ACTION: Add SC's affected by "keyboard interface" to the issue for the definition - @jamieherrera

**Comment by JJdeGroot** [comment]
Discussed in today’s meeting.

<details>
<summary>28 May 2025</summary>

[Source](https://www.w3.org/2025/05/28-matf-minutes.html#e553)

~~~text
JJ: Some issues around “exit methods” may include the common use of on-screen back buttons for navigation, hardware back buttons, Esc key being standard keyboard command.

JJ: Previous discussions of SC 2.1.2, has been around these exit methods, including relation to assistive technology features, such as VoiceOver’s Z gesture (“scrub”, or “escape” gesture)

<JJ> Do the actions listed for 2.1.2 capture everything needed to draft the content?

<Tanya> +1

<Detlev> +1

<Karla> +1

<Joe_Humbert> +1

<pauljadam> There's no WCAG requirement to maintain focus after orientation change

<Detlev> what about 1.3.2?

Detlev: 1.3.2 is certainly the nearest

pauljadam: The orientation change is simply an example scenario explaining the behaviour, not a failure

<pauljadam> On Input or On Focus can be used if focus is lost during those events

pauljadam: True.
~~~
</details>

- ACTION: Group agrees that existing actions capture what's needed to draft 2.1.2

**Comment by JJdeGroot** [comment]
Discussed in MATF meeting on 18 February 2026.

<details>
<summary>Minutes for 18 February 2026</summary>

[Source](https://www.w3.org/2026/02/18-matf-minutes.html#34ac)

~~~text
<JJ> Keyboard interface: [w3c/matf#66 (comment)](https://github.com/w3c/matf/issues/66#issuecomment-3553194380)

<JJ> Accessibility interface: [w3c/matf#247 (comment)](https://github.com/w3c/matf/issues/247#issuecomment-3227673242)

<JJ> Virtual keyboard: [w3c/matf#91 (comment)](https://github.com/w3c/matf/issues/91#issuecomment-3717984460)

quintinb: So the A11y interface is to ensure all assistive tech is represented in the success criteria?

<JJ> https://www.w3.org/TR/wcag2ict-22/#keyboard

shoobe01 It doesn't seem to mean interface in the way he understands it - we need to be clear in what we end up writing. Also, what are these three elements?

shoobe01 keyboards are not the only a11y interface, as dynamic as they are. We should talk about the types of devices

<JJ> Poll: Broaden keyboard interface in 2.1.1 and 2.1.2 to accessibility interface to include more types of assistive technologies?

<shoobe01> +1

<tayef> +1

<Tim> +1

<Joe_Humbert> +1

<JatinV> +1

<quintinb> +1

<pauljadam> 0

<JJ> Results: 6x +1 and 1x 0 -> mostly positive

ACTION: Broaden keyboard interface in 2.1.1 and 2.1.2 to accessibility interface to include more types of assistive technologies

I would like to hear from pauljadam - he's the Keyboard King

pauljadam not sure I'm informed enough to vote, but I think I would keep it as keyboard
~~~
</details>

- ACTION: Broaden keyboard interface in 2.1.1 and 2.1.2 to accessibility interface to include more types of assistive technologies
